### PR TITLE
fix(ui): prevent empty header row from persisting state and crashing CLI

### PR DESCRIPTION
**Pull request by sharan-bruno**
**Jira Link: https://usebruno.atlassian.net/browse/BRU-3228**

### Description

Fixes inconsistent auto-append behavior in the editable table (Headers/Params). Previously a new empty row was only added when typing into the Name field; now it's added when typing into either Name or Value. Also removes stray empty rows: any fully-empty row is dropped, leaving exactly one trailing empty "add" row.

Added a Playwright test covering: typing into Name appends a row, typing only into Value appends a row, and clearing a field removes the now-empty row.
#### Contribution Checklist:

Before:
image

After:
image
image

- [ ] **I've used AI significantly to create this pull request**
- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have added screenshots or gifs to help explain the change if applicable.**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [ ] **Create an issue and link to the pull request.**

Note: Keeping the PR small and focused helps make it easier to review and merge. If you have multiple changes you want to make, please consider submitting them as separate pull requests.

#### Publishing to New Package Managers

Please see [here](../publishing.md) for more information.

## Summary by CodeRabbit

* **Bug Fixes**
  * Editable table add-row behavior refined: trailing empty-row detection now uses any-value logic, ensuring a single trailing empty row when add-row mode is enabled; saving now removes all fully empty rows and edits consistently trigger appending a new empty row.
* **Tests**
  * Added end-to-end test validating auto-append and removal behavior for header rows in the Headers table.
  * Improved test utilities for reliably targeting the last table row during file-upload interactions.

### CODE REVIEW

File: tests/request/headers/auto-append-empty-header-row.spec.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  });
+
+  const headersTable = page.locator('table').first();
+  const rows = headersTable.locator('tbody tr');
+
+  await test.step('Starts with a single empty header row', async () => {
+    await expect(rows).toHaveCount(1);
+  });
+
+  await page.pause();
```
Comment (sanish-bruno) [inline tests/request/headers/auto-append-empty-header-row.spec.ts]: @sharan-bruno remove page.pause()

File: packages/bruno-app/src/components/EditableTable/index.js
Comment (sanish-bruno) [inline packages/bruno-app/src/components/EditableTable/index.js]: maybe redundant now!

With the new filter `updatedRows.filter(hasAnyValue)`, the pushed empty row is immediately filtered out. The `rowsWithEmpty` memo then appends a new empty row on the next render anyway. This push does nothing except allocate a throwaway object and a wasted UUID. The misleading comment ("Only add a new empty row when the key field is filled") also contradicts the PR's goal. Should be removed.

File: packages/bruno-app/src/components/EditableTable/index.js [hunk 2]
```diff
@@ -262,12 +250,10 @@ const EditableTable = ({
...
-    const result = updatedRows.filter((row, i) => {
-      if (showAddRow && i === updatedRows.length - 1) {
-        return hasAnyValue(row);
-      }
-      return true;
-    });
+    // Remove any fully-empty rows from the persisted data. The trailing empty
+    // "add row" is re-added by the rowsWithEmpty memo, so there's always
+    // exactly one empty row at the bottom and never a stray empty row above it.
+    const result = showAddRow ? updatedRows.filter(hasAnyValue) : updatedRows;
```
Comment (sanish-bruno) [inline packages/bruno-app/src/components/EditableTable/index.js:242]: please verify,
**rowsWithEmpty can show duplicate empty rows on initial render**

The removed check prevented appending an empty row when the last row's key field was already empty. Without it, if rows from the parent contains a stray empty row (the exact corrupted state this PR fixes), the memo will append another empty row — producing two empty rows at the bottom on first render.
They'll be cleaned up on the first user interaction, but it's a visible glitch. Consider replacing the removed check with a hasAnyValue-based guard:
```js
  const lastRow = rows[rows.length - 1];
  if (!hasAnyValue(lastRow)) {
    return rows; // last row is already empty, don't add another
  }
```
(This would require extracting hasAnyValue from handleValueChange to a shared utility or a useCallback.)

File: packages/bruno-app/src/components/EditableTable/index.js
Comment (sanish-bruno) [inline packages/bruno-app/src/components/EditableTable/index.js]: isEmptyRow is now inconsistent with hasAnyValue

**Review by sanish-bruno — CHANGES_REQUESTED** [review]

File: packages/bruno-app/src/components/EditableTable/index.js
Comment (sharan-bruno) [reply packages/bruno-app/src/components/EditableTable/index.js]: addressed

File: packages/bruno-app/src/components/EditableTable/index.js (hunk 2 above)
Comment (sharan-bruno) [reply packages/bruno-app/src/components/EditableTable/index.js:242]: Good catch — fixed. Extracted hasAnyValue into a shared useCallback and added the guard to the memo:

Now a stray empty row loaded from a file no longer produces a duplicate — only one empty row renders on initial load. Also unified isEmptyRow to use hasAnyValue so emptiness is defined consistently across the memo guard, the persistence filter, and last-row rendering.

File: packages/bruno-app/src/components/EditableTable/index.js
Comment (sharan-bruno) [reply packages/bruno-app/src/components/EditableTable/index.js]: removed redundant code

File: tests/request/headers/auto-append-empty-header-row.spec.ts (hunk 1 above)
Comment (sharan-bruno) [reply tests/request/headers/auto-append-empty-header-row.spec.ts]: removed page.pause()

File: tests/utils/page/actions.ts [hunk 3]
```diff
@@ -1118,13 +1118,14 @@ const addMultipartFileToLastRow = async (page: Page, electronApp: ElectronApplic
...
 
     const table = buildCommonLocators(page).table('editable-table');
-    const lastRow = table.allRows().last();
-
-    await expect(lastRow.locator('.upload-btn')).toBeVisible();
-    await lastRow.locator('.upload-btn').click();
-    await expect(lastRow.locator('.file-value-cell')).toBeVisible();
-    const inlineChip = lastRow.getByTestId('multipart-file-chip').filter({ hasText: path.basename(filePath) });
-    const summary = lastRow.getByTestId('multipart-file-summary');
+    const rowIndex = (await table.allRows().count()) - 1;
```
Comment (sanish-bruno) [inline tests/utils/page/actions.ts:1124]: please add a comment on why we are choosing row - 1

File: packages/bruno-app/src/components/EditableTable/index.js [hunk 4]
```diff
@@ -177,16 +188,11 @@ const EditableTable = ({
...
-      const lastRowKeyValue = keyColumn.getValue ? keyColumn.getValue(lastRow) : lastRow[keyColumn.key];
-      const isLastRowEmpty = !lastRowKeyValue || (typeof lastRowKeyValue === 'string' && lastRowKeyValue.trim() === '');
-
-      if (isLastRowEmpty) {
-        return rows;
-      }
+    // If the last row is already empty (e.g. a stray empty row loaded from a
+    // pre-existing file), don't append another one — otherwise the table would
+    // render two empty rows at the bottom on the initial render.
+    if (!hasAnyValue(rows[rows.length - 1])) {
```
Comment (sanish-bruno) [inline packages/bruno-app/src/components/EditableTable/index.js:194]: hasAnyValue treats false as "no value" , boolean false is treated as "no value." , disabled check box can cause issues, please check

File: tests/request/headers/auto-append-empty-header-row.spec.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    await expect(rows).toHaveCount(3);
+  });
+
+  const selectAll = process.platform === 'darwin' ? 'Meta+a' : 'Control+a';
+
+  await test.step('Clearing the Name field empties that row and removes it', async () => {
+    // Row 0 has only a Name ("Content-Type"); clearing it makes the row fully
+    // empty, so it is dropped and the trailing empty row remains.
+    const nameEditor = rows.first().locator('[data-testid="column-name"] .CodeMirror');
```
Comment (sanish-bruno) [inline tests/request/headers/auto-append-empty-header-row.spec.ts]: is is possible to use getByTestId here instead of `.locator('[data-testid="column-name"] .CodeMirror')`, applicable to rest of the places too, whereever it is applicable

**Review by sanish-bruno — CHANGES_REQUESTED** [review]

File: tests/request/headers/auto-append-empty-header-row.spec.ts (hunk 5 above)
Comment (sharan-bruno) [reply tests/request/headers/auto-append-empty-header-row.spec.ts]: addressed

File: packages/bruno-app/src/components/EditableTable/index.js (hunk 4 above)
Comment (sharan-bruno) [reply packages/bruno-app/src/components/EditableTable/index.js:194]: working fine with all the edge cases

File: tests/utils/page/actions.ts (hunk 3 above)
Comment (sharan-bruno) [reply tests/utils/page/actions.ts:1124]: added a comment on top of it

**Review by bijin-bruno — APPROVED** [review]
